plugins/modules/source_control/gitlab/gitlab_service.py

```diff
@@ -0,0 +1,468 @@
+version_added: '2.4.0'
+description:
+  - Creates, updates, or deletes GitLab integrations formerly known as "services".
+author:
+  - Raphaël Droz (@drzraf)
+requirements:
+  - python-gitlab python module
+extends_documentation_fragment:
+- community.general.auth_basic
+
+options:
+  api_token:
+    description:
+      - GitLab token for logging in.
+    type: str
+  project:
+    description:
+      - The ID or path of the project (urlencoded or not).
+    required: true
+    type: str
+  active:
+    description:
+      - Not yet supported (U(https://gitlab.com/gitlab-org/gitlab-ce/issues/41113)).
+    type: bool
+    default: yes
+  service:
+    description:
+      - The type of service.
+    required: true
+    type: str
+    choices:
+      - asana
+      - assembla
+      - bamboo
+      - bugzilla
+      - buildkite
+      - campfire
+      - confluence
+      - custom-issue-tracker
+      - datadog
+      - discord
+      - drone-ci
+      - emails-on-push
+      - ewm
+      - external-wiki
+      - flowdock
+      - hangouts-chat
+      - hipchat
+      - irker
+      - jenkins
+      - jira
+      - mattermost
+      - mattermost-slash-commands
+      - microsoft-teams
+      - packagist
+      - pipelines-email
+      - pivotaltracker
+      - prometheus
+      - pushover
+      - redmine
+      - slack
+      - slack-slash-commands
+      - teamcity
+      - unify-circuit
+      - webex-teams
+      - youtrack
+  params:
+    description:
+      - The description of the service, see documentation U(https://docs.gitlab.com/ee/api/services.html).
+    type: dict
+  events:
+    description:
+      - The events that trigger the service.
+      - Required if I(state=present).
+    choices: ["push", "issues", "confidential_issues", "merge_requests", "tag_push", "note", "confidential_note", "job", "pipeline", "wiki_page"]
+    default: ["push", "issues", "confidential_issues", "merge_requests", "tag_push", "note", "confidential_note", "job", "pipeline", "wiki_page"]
+    type: list
+    elements: str
+  state:
+    description:
+      - Create or delete a service.
+      - Possible values are C(present) and C(absent).
+    default: "present"
+    choices: ["present", "absent"]
+    type: str
+'''
+
+EXAMPLES = '''
+# Setup email on push for this project
+- name: emails me on push
+  community.general.gitlab_service:
+    api_url: https://gitlab.com
+    api_token: foobar
+    project: 123456
+    service: emails-on-push
+    params:
+      recipients: foo@example.com
+      disable_diffs: true
+    state: present
+  delegate_to: localhost
+
+# This will always be set to change because a non-null token is mandatory
+- name: trigger packagist update on push events (only)
+  community.general.gitlab_service:
+    api_url: https://gitlab.com
+    api_token: foobar
+    project: foo/proj
+    service: packagist
+    events: ["push"]
+    params:
+      username: foo
+      token: bar
```

Comment:
Since this is a secret, the task should be marked `no_log` to avoid this parameter ending up in logs.

Reply:
I found a better way, in the last commit I set individual `no_log` value for token/api_key/... parameters of individual services.

Reply:
That does not work. The first `AnsibleModule` invocation will dump all these fields to syslog.

Reply:
Good point.